src/commands/generate.ts

```diff
@@ -54,22 +54,22 @@ export default class Generate extends Command {
 		const { flags } = this.parse(Generate);
 
 		// get our directory context correct
-		const getRepoDirectory = (p: string) => {
-			if (fs.statSync(p).isDirectory()) {
+		const getEnvironmentDirectory = async (p: string) => {
+			if ((await fs.stat(p)).isDirectory()) {
 				return p;
 			}
 
 			return path.dirname(p);
 		};
 
-		const repoDir = getRepoDirectory(flags.environmentPath);
-		const productDir = path.join(repoDir, 'product');
+		const environmentDir = await getEnvironmentDirectory(flags.environmentPath);
+		const productDir = path.join(environmentDir, 'product');
 
 		// find the manifests to use...
 		const manifestFiles = [
 			'config-manifest.yml', // product-specific manifest
 			`deploy/${flags.target}/config-manifest.yml`, // target-specific manifest
-		].filter(p => fs.existsSync(path.join(productDir, p)));
+		].filter(async p => await fs.exists(path.join(productDir, p)));
```

Comment:
I've got a feeling this will keep everything in `manifestFiles` regardless, since `Array.prototype.filter` is unaware of promises it'll just cast the Promise to a boolean which I think results in `true` - You can use `await Bluebird.filter([], fn) as an easy way to fix this

Reply:
Also it might be worth adding a test for this

Reply:
Thanks @Page- I have pushed the change, and will try to think of a way to test this...